src/nwb_datajoint/spikesorting/spikesorting_curation.py

```diff
@@ -498,53 +604,45 @@ def make(self, key):
         # keep the quality metrics only if no merging occurred.
         metrics = quality_metrics if not units_merged else None
 
-        # save the new curation
-        curated_sorting_name = self._get_curated_sorting_name(key)
-        key['sorting_path'] = str(
-            Path(os.getenv('NWB_DATAJOINT_SORTING_DIR')) /
-            Path(curated_sorting_name))
-        if os.path.exists(key['sorting_path']):
-            shutil.rmtree(key['sorting_path'])
-        parent_sorting = parent_sorting.save(folder=key['sorting_path'])
-
         # insert this sorting into the CuratedSpikeSorting Table
-        key['sorting_id'] = Curation.insert_curation(
-            key, parent_curation_id=key['parent_curation_id'],
-            labels=labels, merge_groups=merge_groups, metrics=metrics,
-            description='auto curated')
-        self.insert1(key)
+        # first remove keys that aren't part of the Sorting (the primary key of curation)
+        c_key = (SpikeSorting & key).fetch("KEY")[0]
+        curation_key = {item: key[item] for item in key if item in c_key}
+        key['auto_curation_key'] = Curation.insert_curation(
+            curation_key, parent_curation_id=parent_curation_id,
+            labels=labels, merge_groups=merge_groups, metrics=metrics, description='auto curated')
 
-    def fetch_nwb(self, *attrs, **kwargs):
-        return fetch_nwb(self, (AnalysisNwbfile, 'analysis_file_abs_path'),
-                         *attrs, **kwargs)
+        self.insert1(key)
 
     @staticmethod
-    def get_merge_groups(sorting, parent_merge_groups, quality_metrics,
+    def get_merge_groups(sorting,
+                         parent_merge_groups,
+                         quality_metrics,
                          merge_params):
         """Identifies units to be merged based on the quality_metrics and
-        merge parameters and returns an updated list of merges for the curation
+        merge parameters and returns an updated list of merges for the curation.
 
         Parameters
-        ----------
-        sorting : spike interface sorting object
+        ---------
+        sorting : spikeinterface.sorting
         parent_merge_groups : list
             Information about previous merges
-        quality_metrics : dict
-            quality metrics by unit
+        quality_metrics : list
         merge_params : dict
 
         Returns
         -------
-        sorting : sorting_extractor
+        merge_groups : list of lists
         merge_occurred : bool
 
         """
+
         # overview:
         # 1. Use quality metrics to determine merge groups for units
         # 2. Combine merge groups with current merge groups to produce union of merges
 
         if not merge_params:
-            return sorting, False
+            return parent_merge_groups, False
         else:
             # TODO: use the metrics to identify clusters that should be merged
```

Comment:
here `new_merges` is defined as an empty list, and below this line there is a for loop that goes through the elements of `new_merges` (which don't exist) and later we index into the element in line 653. is this intended? it's unclear to me how the code would behave here.